addons/mail/static/src/models/autocomplete_input_view.js

```diff
@@ -17,7 +38,58 @@ registerModel({
                 return;
             }
         },
-         /**
+        /**
+         * @private
+         * @param {MouseEvent} ev
+         */
+        onKeydown(ev) {
+            if (ev.key === 'Escape') {
+                this.onBlur();
+            }
+        },
+        /**
+         * @private
+         * @param {MouseEvent} ev
+         */
+        onSelect(ev, ui) {
+            if (this.chatWindowOwnerAsNewMessage) {
+                this.chatWindowOwnerAsNewMessage.onAutocompleteSelect(ev, ui);
+                return;
+            }
+            if (this.discussSidebarCategoryOwnerAsAddingItem) {
+                this.discussSidebarCategoryOwnerAsAddingItem.onAddItemAutocompleteSelect(ev, ui);
+                return;
+            }
+            if (this.discussViewOwnerAsMobileAddItemHeader) {
+                this.discussViewOwnerAsMobileAddItemHeader.onMobileAddItemHeaderInputSelect(ev, ui);
+                return;
+            }
+            if (this.messagingMenuOwnerAsMobileNewMessageInput) {
+                this.messagingMenuOwnerAsMobileNewMessageInput.onMobileNewMessageInputSelect(ev, ui);
+                return;
+            }
+        },
+        /**
+         * @private
+         * @param {Object} req
+         * @param {string} req.term
+         * @param {function} res
+         */
+        onSource(req, res) {
+            if (this.chatWindowOwnerAsNewMessage) {
+                this.chatWindowOwnerAsNewMessage.onAutocompleteSource(req, res);
+            } else if (this.discussSidebarCategoryOwnerAsAddingItem) {
+                this.discussSidebarCategoryOwnerAsAddingItem.onAddItemAutocompleteSource(req, res);
+            } else if (this.discussViewOwnerAsMobileAddItemHeader) {
+                this.discussViewOwnerAsMobileAddItemHeader.onMobileAddItemHeaderInputSource(req, res);
+            } else if (this.messagingMenuOwnerAsMobileNewMessageInput) {
+                this.messagingMenuOwnerAsMobileNewMessageInput.onMobileNewMessageInputSource(req, res);
+            }
+            if (this.messaging) {
+                this.messaging.messagingBus.trigger('o-AutocompleteInput-source', {});
+            }
```

Comment:
This only seems to be useful in tests so perhaps it shouldn't be there but rather added via a patch in test assets. It will also allow you to get rid of this else-if forest and thus achieve symmetry with other handlers.